qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -249,10 +264,20 @@ def compute_eigenvalues(
         prev_states = []
 
         for step in range(1, self.k + 1):
+            if isinstance(self.initial_point, list):
+                initial_point = validate_initial_point(self.initial_point[step - 1], self.ansatz)
+
+            else:
+                initial_point = validate_initial_point(self.initial_point, self.ansatz)
 
-            # update list of optimal circuits
             if step > 1:
-                prev_states.append(self.ansatz.bind_parameters(result.optimal_points[-1]))
+                if isinstance(self.initial_point, list):
+                    params = validate_initial_point(self.initial_point[step - 1], self.ansatz)
```

Comment:
If I understood VQD correctly, our cost function for parameters `x`, an ansatz `phi` is
```
L(x) = <phi(x)|H|phi(x)> + penalty * ( |<phi(groundstate)|phi(x)>|^2 + |<phi(1st_excited)|phi(x)>|^2 + ...)
```
where `groundstate` are the parameters for the ground state and `1st_excited` for the first excited etc. So the `prev_states` variable here I think should always be the optimal points.

The initial points in Qiskit are usually understood as starting points for the optimizations, so at iteration `k` of the VQD for the `k-th` excited state, element `k` of the `initial_points` list should be the initial point of the optimization. Instead here, I think you are implementing
```
L(x) = <phi(x)|H|phi(x)> + penalty * ( |<phi(initial_points[0])|phi(x)>|^2 + |<phi(initial_points[1])|phi(x)>|^2 + ...)
```
or maybe I misunderstood the code 🤔 

If you indeed need the latter, then we can also discuss adding that, but I think the `initial_points` should only act as starting points for the optimization and not be part of the loss function.

Reply:
Thanks for the comment. Yes you are right, what I want to do is use the previous set of parameters as in `L(x) = <phi(x)|H|phi(x)> + penalty * ( |<phi(groundstate)|phi(x)>|^2 + |<phi(1st_excited)|phi(x)>|^2 + ...)` but put in `x` a given set of parameters from the list of the initial points and then do the optimization. The local file that I was working on was slightly different from the version that I then modified and uploaded. I have to check better on this.